src/main/java/uk/gov/laa/ccms/caab/service/ApplicationService.java

```diff
@@ -425,6 +434,62 @@ public AddressFormData getCorrespondenceAddressFormData(final String id) {
         .map(addressFormDataMapper::toAddressFormData).block();
   }
 
+  /**
+   * Retrieves linked cases associated with a specific ID.
+   * This method fetches a list of linked cases, transforms them into a
+   * {@code ResultsDisplay<LinkedCaseResultRowDisplay>} format, and returns the result.
+   *
+   * @param id The unique identifier for the cases to be retrieved.
+   * @return A {@code ResultsDisplay<LinkedCaseResultRowDisplay>} containing the linked cases.
+   */
+  public ResultsDisplay<LinkedCaseResultRowDisplay> getLinkedCases(final String id) {
+    final ResultsDisplay<LinkedCaseResultRowDisplay> results = new ResultsDisplay<>();
+
+    return caabApiClient.getLinkedCases(id)
+        .flatMapMany(Flux::fromIterable) // Convert to Flux<LinkedCase>
+        .map(resultDisplayMapper::toLinkedCaseResultRowDisplay) // Map to ResultRowDisplay
+        .collectList() // Collect into a List
+        .map(list -> {
+          results.setContent(list); // Set the content of ResultsDisplay
+          return results; // Return the populated ResultsDisplay
+        }).block();
+  }
+
+  /**
+   * Removes a linked case from a primary case.
+   * This method communicates with the CAAB API client to un-link a case identified by
+   * {@code linkedCaseId} from a primary case identified by {@code id}.
+   *
+   * @param id           The ID of the primary case from which the linked case will be removed.
+   * @param linkedCaseId The ID of the linked case to be removed.
+   * @param user         The user performing the operation, identified by {@code UserDetail}.
+   */
+  public void removeLinkedCase(final String id, final String linkedCaseId, final UserDetail user) {
```

Comment:
Change id to applicationId (or primaryCaseId) or something.

Reply:
done